Stratis.Bitcoin.Features.LightWallet/LightWalletSyncManager.cs

```diff
@@ -87,24 +102,53 @@ public void Start()
                 // we're looking from where to start syncing the wallets.
                 // we start by looking at the heights of the wallets and we start syncing from the oldest one (the smallest height).
                 // if for some reason we can't find a height, we look at the creation date of the wallets and we start syncing from the earliest date.
-                int? earliestWalletHeight = this.walletManager.Wallets.Min(w => w.AccountsRoot.Single(a => a.CoinType == this.coinType).LastBlockSyncedHeight);
+                int? earliestWalletHeight = this.walletManager.GetEarliestWalletHeight();
                 if (earliestWalletHeight == null)
                 {
-                    DateTimeOffset oldestWalletDate = this.walletManager.Wallets.Min(w => w.CreationTime);
+                    DateTimeOffset oldestWalletDate = this.walletManager.GetOldestWalletCreationTime();
+
+                    if (fork != null && oldestWalletDate > fork.Header.BlockTime)
```

Comment:
I need @dangershony or @Aprogiena to have a look at this and the change just below.

Reply:
I am just wondering why do we need a new parameter called `fork`, using the `this.walletTip` should yield the same result right?

Reply:
I would leave this to @dangershony as I'm not that familiar with this part of code base. It would take me quite a long time to learn it. But if you really need me, ping me again.